hooks/DirectX.cpp

```diff
@@ -361,6 +361,18 @@ HRESULT __stdcall dPresent(IDXGISwapChain* __this, UINT SyncInterval, UINT Flags
         ImGuiRenderer::Submit([]() { Replay::Render(); });
     }
 
+    // TODO: teleport test
+    if (State.Teleport && ImGui::IsMouseClicked(ImGuiMouseButton_Right)) {
```

Comment:
why is this feature added into the directX layer ? wasn't there a possibility to add this into one of the hooked update functions ? please move this somewhere else like in playercontrol update maybe